tests/src/test/scala/org/apache/openwhisk/core/containerpool/test/ContainerProxyTests.scala

```diff
@@ -365,18 +372,18 @@ class ContainerProxyTests
       container.resumeCount shouldBe 1
       acker.calls should have size 2
 
-      // Add debugging information for the case of failure.
-      println(acker.calls.map(_._2.toJson.compactPrint))
-
       store.calls should have size 2
-      acker
-        .calls(0)
-        ._2
-        .annotations
+
+      // As the active acks are sent asynchronously, it is possible, that the activation with the init time is not the
+      // first one in the buffer.
+      val initializedActivations =
+        acker.calls.filter(_._2.annotations.get(WhiskActivation.initTimeAnnotation).isDefined)
+      initializedActivations should have size 1
+
+      initializedActivations.head._2.annotations
```

Comment:
For the instances where you only need one of the calls:

```scala
val initRunActivation = acker.calls.find(_._2.annotations.get(WhiskActivation.initTimeAnnotation).isDefined).getOrElse(fail("..."))
```

?

Reply:
This would keep consistency to the other tests that I've adapted, but then we need another check to verify that only one of the activations contains the init annotation.
So personally I'd prefer to use `filter`, checking the size of the result and using head.